Code/src/main/java/nl/utwente/group10/haskell/type/ConstT.java

```diff
@@ -69,4 +70,26 @@ public String toString() {
             return this.constructor;
         }
     }
+
+    @Override
+    public int compareTo(final Type type) {
+        if (type instanceof ConstT) {
```

Comment:
If you do a quick `ConstT constt = (ConstT) type;` here you can avoid a few casts later.